radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/CreateTokenToParticleGroupsMapper.java

```diff
@@ -45,27 +46,27 @@
 		}
 
 		TokenDefinitionParticle token = new TokenDefinitionParticle(
-				tokenCreation.getAddress(),
+			tokenCreation.getAddress(),
 			tokenCreation.getName(),
-				tokenCreation.getIso(),
-				tokenCreation.getDescription(),
-				tokenCreation.getGranularity(),
-				ImmutableMap.of(
-					MintedTokensParticle.class, mintPermissions,
-					BurnedTokensParticle.class, burnPermissions,
-					TransferredTokensParticle.class, TokenPermission.ALL
-				),
-				null
+			tokenCreation.getIso(),
+			tokenCreation.getDescription(),
+			TokenUnitConvert.unitsToSubunits(tokenCreation.getGranularity()),
+			ImmutableMap.of(
+				MintedTokensParticle.class, mintPermissions,
+				BurnedTokensParticle.class, burnPermissions,
+				TransferredTokensParticle.class, TokenPermission.ALL
+			),
+			null
 		);
 
-		if (tokenCreation.getInitialSupply().isZero()) {
+		if (tokenCreation.getInitialSupply().compareTo(BigDecimal.ZERO) <= 0) {
```

Comment:
I think a negative initial supply should be an error, not silently truncated to zero.

Reply:
yikes, thanks for catching that